components/ArticleCard.js

```diff
@@ -1,19 +1,89 @@
 import styled from "styled-components";
-import Link from "next/link";
+import { useRouter } from "next/router";
+import { useStore } from "../hooks/useStore";
 
-export default function ArticleCard({ id, text }) {
+export default function ArticleCard({
+  id,
+  url,
+  urlToImage,
+  title,
+  author,
+  description,
+  message,
+  alert,
+}) {
+  const router = useRouter();
+  const setArticle = useStore((state) => state.setArticle);
+
+  async function getFullContent() {
+    const response = await fetch(`/api/search/${id}`, {
+      method: "POST",
```

Comment:
Post is not the correct request method here. You use post if you want to create a new resource. You are simply fetching data here. This would be a GET request.

Reply:
I know it should be a GET request. I just use POST as a convenient way to get data (actually parameters) from the request.body and then manipulate the params here. But yeah i should change it to GET and get the parameters from request.query after "polishing" parameters from the client side. 
Thank you!

Reply:
Really? I can not make a fetch request from irgendein component? It s not the best practice?

Reply:
Any component can fetch, that is not the point. It's just that you are using the zustand store to temporarily store data that is specific for one article. Then you navigate to the article page and the url `localhost:3000/articleid` highly suggests that you can use this url again to retrieve the article specific site. But that does not work. The site always shows what is currently in the zustand store.

Reply:
The site that you reach under that specific url should always show the article specific data, or the url should not be specific